crates/fuel-gas-price-algorithm/src/v1.rs

```diff
@@ -355,6 +350,9 @@ impl AlgorithmUpdaterV1 {
 
             // metadata
             self.unrecorded_blocks.insert(height, block_bytes);
+            self.unrecorded_blocks_bytes = self
+                .unrecorded_blocks_bytes
+                .saturating_add(block_bytes as u128);
```

Comment:
this leaves me with a question. Can we add a block twice, e.g. in the case of a reorganization? If so it might be necessary to subtract the `unrecorded_block_bytes` for the block that is already present at the same height.

Reply:
If the reorg happens on L2, then we will roll back the state of the algorithm too. We regularly store the `GasPriceMetadata` to our database so we can return to previous state. That only happens when we _restart_ the client, it populates the algorithm values with the metadata from that height.

Reply:
Thanks, really clear explanation :D

Reply:
if we just get finalized data we dont need to worry about reorgs too much